File: src/main/kotlin/org/kite9/diagram/common/elements/factory/DiagramElementFactory.java
package org.kite9.diagram.common.elements.factory;

import org.kite9.diagram.model.DiagramElement;

/**
 * Allows for the conversion from an `XMLElement` to a Kite9 `DiagramElement` (used in layout).
 * 
 * @author robmoffat
 *
 */
public interface DiagramElementFactory<X> {

	public DiagramElement createDiagramElement(X in, DiagramElement parent);

	public TemporaryConnected createTemporaryConnected(DiagramElement parent, String idSuffix);
	
}

File: src/main/kotlin/org/kite9/diagram/common/elements/factory/DiagramElementFactory.kt
package org.kite9.diagram.common.elements.factory

import org.kite9.diagram.model.DiagramElement

/**
 * Allows for the conversion from an `X` (usually XML element) to a Kite9 `DiagramElement` (used in layout).
 *
 * @author robmoffat
 */
interface DiagramElementFactory<X> {

    fun createDiagramElement(x: X, parent: DiagramElement): DiagramElement

    fun createTemporaryConnected(parent: DiagramElement, idSuffix: String): TemporaryConnected

}